bot-tezos-swap/library/util.js

```diff
@@ -0,0 +1,92 @@
+const { BigNumber } = require("bignumber.js");
+
+const config = require(`./${
+  process.env.BOT_ENV || "prod"
+}-network-config.json`);
+
+/**
+ * Returns the minimum expected value by an initiator after deducting reward for swap responder
+ *
+ * @param swapValue the actual swap value initiated by the user
+ * @param rewardInBIPS reward taken by the swap responder in basis points
+ */
+module.exports.calcSwapReturn = (swapValue, rewardInBIPS) =>
+  new BigNumber(swapValue)
+    .multipliedBy(
+      new BigNumber(1).minus(new BigNumber(rewardInBIPS).div(10000))
+    )
+    .toFixed(0, 3);
+
+/**
+ * constants for calculations
+ * feePad : contains the fee padding of each asset for each pair tx fee [Y*txFee]
+ * minTradeVolume : contains minimum volume of swaps expected of each asset for each pair, helps calculate estimated eth balance req. (no. of swaps = maxVolume/minVolume)*feePerTx*feePad
+ */
+module.exports.constants = {
+  //   feePad: {
+  //     "usdc/usdtz": {
+  //       usdc: new BigNumber(config.pairs["usdc/usdtz"]["usdc"].feePad),
+  //       usdtz: new BigNumber(config.pairs["usdc/usdtz"]["usdtz"].feePad),
+  //     },
+  //     "eth/ethtz": {
+  //       eth: new BigNumber(config.pairs["eth/ethtz"]["eth"].feePad),
+  //       ethtz: new BigNumber(config.pairs["eth/ethtz"]["ethtz"].feePad),
+  //     },
+  //   },
+  minTradeVolume: {
+    "xtz/usdtz": {
```

Comment:
should make this dynamic in the following release

Reply:
This is more of a worst case estimate, as long as the value is small enough I think that should be fine